referenceframe/frame.go

```diff
@@ -31,14 +31,17 @@ type Limit struct {
 
 // RestrictedRandomFrameInputs will produce a list of valid, in-bounds inputs for the frame, restricting the range to
 // `lim` percent of the limits.
-func RestrictedRandomFrameInputs(m Frame, rSeed *rand.Rand, lim float64) []Input {
+func RestrictedRandomFrameInputs(m Frame, rSeed *rand.Rand, lim float64, nodeLocation []Input) ([]Input, error) {
```

Comment:
Per our discussion offline, lets rename `restrictionPercent` and `reference`.  In addition the comment above this function should be modified to capture these changes